nncf/experimental/torch/nas/bootstrapNAS/elasticity/onnx_export.py

```diff
@@ -0,0 +1,94 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from typing import List
+
+import torch
+from torch import nn
+from torch.utils.hooks import RemovableHandle
+
+from nncf.torch.exporter import PTExporter
+
+
+class BNTrainingStateSwitcher:
```

Comment:
Can we keep this in the main torch code? We might need this later, I don't think that this is only specific to NAS by definition (although I do see that currently will only be used for NAS).

Reply:
Don't see when it's needed now. It's specific implementation with hooks that is needed for export in training.
BNAdaptation uses different implementation.